src/util/ChartUtils.ts

```diff
@@ -437,27 +380,41 @@ export const getBarPosition = ({
   return result;
 };
 
-export const appendOffsetOfLegend = (offset: any, items: Array<FormattedGraphicalItem>, props: any, legendBox: any) => {
+export const appendOffsetOfLegend = (
+  offset: ChartOffset,
+  _unused: unknown,
+  props: {
+    width?: number;
+    margin: Margin;
+    children?: ReactNode[];
+  },
+  legendBox: DOMRect | null,
+): ChartOffset => {
   const { children, width, margin } = props;
   const legendWidth = width - (margin.left || 0) - (margin.right || 0);
-  // const legendHeight = height - (margin.top || 0) - (margin.bottom || 0);
   const legendProps = getLegendProps({ children, legendWidth });
-  let newOffset = offset;
-
   if (legendProps) {
-    const box = legendBox || {};
+    const { width: boxWidth, height: boxHeight } = legendBox || {};
     const { align, verticalAlign, layout } = legendProps;
 
-    if ((layout === 'vertical' || (layout === 'horizontal' && verticalAlign === 'middle')) && isNumber(offset[align])) {
-      newOffset = { ...offset, [align]: newOffset[align] + (box.width || 0) };
+    if (
+      (layout === 'vertical' || (layout === 'horizontal' && verticalAlign === 'middle')) &&
+      align !== 'center' &&
```

Comment:
is this a new condition? why?

Reply:
To satisfy typescript - without this condition, TS was flagging (correctly) that `offset` is never going to have property `center`.

Reply:
Gotcha, cool. As long as the logic shakes out to be the same